### PR TITLE
Nn 2589 new scheduled activity endpoint

### PR DESCRIPTION
**Pull request by JamieDraperUK**

### CODE REVIEW

File: src/main/java/net/syscon/elite/api/resource/ScheduleResource.java [hunk 1]
```diff
@@ -96,15 +105,22 @@
...
                                          @ApiParam(value = "The required offender numbers (mandatory)", required = true) @RequestBody List<String> body,
                                          @ApiParam(value = "Date of whereabouts list, default today") @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) @RequestParam(value = "date", required = false) LocalDate date,
                                          @ApiParam(value = "AM, PM or ED", allowableValues = "AM,PM,ED") @RequestParam(value = "timeSlot", required = false) TimeSlot timeSlot,
                                          @ApiParam(value = "Whether to include 'excluded' activities in the results", defaultValue = "false") @RequestParam(value = "includeExcluded", required = false, defaultValue = "false") boolean includeExcluded);
 
+
+    @PostMapping("/activities")
+    @ApiOperation(value = "", nickname = "getActivities")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "", response = PrisonerSchedule.class, responseContainer = "List")})
```
Comment (petergphillips) [inline src/main/java/net/syscon/elite/api/resource/ScheduleResource.java:121]: unnecessary 200 - will be generated automatically (esp. as aren't setting a custom message)

File: src/main/java/net/syscon/elite/api/resource/ScheduleResource.java [hunk 2]
```diff
@@ -96,15 +105,22 @@
...
+    List<PrisonerSchedule> getActivities(@ApiParam(value = "Event ids(mandatory)", required = true) @RequestBody List<Long> body);
```
Comment (petergphillips) [inline src/main/java/net/syscon/elite/api/resource/ScheduleResource.java:122]: do you need a `NotEmpty` or other annotation here to enforce the mandatory nature?  Can just see it returning no results if the body isn't passed in.  Should it be a 400?  Could do with a resource test for that...

File: src/main/java/net/syscon/elite/api/resource/ScheduleResource.java (hunk 1 above)
Comment (JamieDraperUK) [reply src/main/java/net/syscon/elite/api/resource/ScheduleResource.java:121]: The code seems to be a required parameter of ApiResponse. What did you have in mind? Happy to change this and the rest in this file.

File: src/main/java/net/syscon/elite/api/resource/ScheduleResource.java (hunk 2 above)
Comment (JamieDraperUK) [reply src/main/java/net/syscon/elite/api/resource/ScheduleResource.java:122]: Will add one now.

File: src/main/java/net/syscon/elite/api/resource/impl/SchedulesResourceImpl.java [hunk 3]
```diff
@@ -76,10 +76,15 @@ public SchedulesResourceImpl(final SchedulesService schedulesService, final Appo
...
 
     @Override
-    public List<PrisonerSchedule> getActivities(final String agencyId, final List<String> body, final LocalDate date, final TimeSlot timeSlot, final boolean includeExcluded) {
+    public List<PrisonerSchedule> getActivitiesForBookings(final String agencyId, final List<String> body, final LocalDate date, final TimeSlot timeSlot, final boolean includeExcluded) {
         return schedulesService.getActivities(agencyId, body, date, timeSlot, includeExcluded);
     }
 
+    @Override
+    public List<PrisonerSchedule> getActivities(final List<Long> eventIds) {
+        return schedulesService.getActivities(eventIds);
```
Comment (petergphillips) [inline src/main/java/net/syscon/elite/api/resource/impl/SchedulesResourceImpl.java:85]: can't remember how does security work for this - will anyone with a token be able to access everything?

Comment (JamieDraperUK) [reply src/main/java/net/syscon/elite/api/resource/impl/SchedulesResourceImpl.java:85]: Good catch! Not sure the schedule is too sensitive but will verify the agency like the other methods in the service. At least then it's consistent.

**Pull request merged**